scripts/release/generate-changelog.js

```diff
@@ -73,53 +75,56 @@ function getChangesList() {
   const lastTagName = command`git describe --tags ${lastTagHash}`.run()
 
   const commits = command`git log ${lastTagName.trimEnd()}..HEAD --pretty=format:%s`.run()
-
   const changesWithEmojis = emojiNameToUnicode(commits)
 
-  const allowedChanges = changesWithEmojis
-    .split('\n')
-    .filter(isNotVersionEntry)
-    .filter(isNotMaintenanceEntry)
-    .sort(byEmojiPriority)
+  let changes = changesWithEmojis.split('\n').filter(isNotVersionEntry)
+  let internalChanges = []
+  let publicChanges = []
+
+  changes.forEach((entry) => {
+    let trimmedEntry = entry.trim()
+    if (INTERNAL_EMOJI_PRIORITY.some((emoji) => trimmedEntry.startsWith(emoji))) {
+      internalChanges.push(entry)
+    } else {
+      publicChanges.push(entry)
+    }
+  })
+
+  internalChanges = internalChanges
+    .sort((a, b) => sortByEmojiPriority(a, b, INTERNAL_EMOJI_PRIORITY))
+    .map((entry) => `- ${entry}`)
+  publicChanges = publicChanges
+    .sort((a, b) => sortByEmojiPriority(a, b, PUBLIC_EMOJI_PRIORITY))
     .map((entry) => `- ${entry}`)
-    .join('\n')
 
-  // changes with pull request links
-  return allowedChanges.replace(
-    /\(#(\d+)\)/gm,
-    (_, id) => `([#${id}](https://github.com/DataDog/browser-sdk/pull/${id}))`
-  )
-}
+  return `
+**Public Changes:**
+${publicChanges.join('\n')}
 
-function isNotVersionEntry(line) {
-  return !/^v\d+\.\d+\.\d+/.test(line)
+**Internal Changes:**
+${internalChanges.join('\n')}
+`.replace(/\(#(\d+)\)/gm, (_, id) => `([#${id}](https://github.com/DataDog/browser-sdk/pull/${id}))`)
 }
 
-function isNotMaintenanceEntry(line) {
-  return !/^👷/.test(line)
+function sortByEmojiPriority(a, b, priorityList) {
+  const getFirstRelevantEmoji = (text) => {
+    const matches = text.match(EMOJI_REGEX) || []
+    return matches.find((emoji) => priorityList.includes(emoji))
+  }
+
+  const emojiA = getFirstRelevantEmoji(a)
+  const emojiB = getFirstRelevantEmoji(b)
+  const indexA = emojiA ? priorityList.indexOf(emojiA) : Number.MAX_VALUE
+  const indexB = emojiB ? priorityList.indexOf(emojiB) : Number.MAX_VALUE
+
+  return indexA - indexB
 }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏It can be simplified:
```suggestion
function sortByEmojiPriority(a, b, priorityList) {
  const getFirstRelevantEmojiIndex = (text) => {
    const matches = text.match(EMOJI_REGEX) || [];
    const emoji = matches.find((emoji) => priorityList.includes(emoji));
    return emoji ? priorityList.indexOf(emoji) : Number.MAX_VALUE;
  };

  return getFirstRelevantEmojiIndex(a) - getFirstRelevantEmojiIndex(b);
}
```